frappe/boot.py

```diff
@@ -151,13 +151,11 @@ def get_allowed_report_names(cache=False) -> set[str]:
 
 def get_user_pages_or_reports(parent, cache=False):
 	if cache:
-		has_role = frappe.cache.get_value("has_role:" + parent, user=frappe.session.user)
+		has_role = frappe.cache.get_value(f"has_role:{parent}", user=frappe.session.user)
```

Comment:
Conversions like these are scary if tests don't cover all use cases. If parent passed is None, an error would've been raised which might've been handled somewhere up the stack or at least it's a solid indicator that something is breaking. Whereas, changing this to an f-string will just evaluate it to "has_role:None"